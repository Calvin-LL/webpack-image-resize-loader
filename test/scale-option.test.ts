import webpack from "webpack";

import compile from "./helpers/compile";
import execute from "./helpers/execute";
import getCompiler from "./helpers/getCompiler";
import readAsset from "./helpers/readAsset";

describe('"scale" option', () => {
  it("should work with 1", async () => {
    const compiler = getCompiler({
      scale: 1,
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it("should work with 0.5", async () => {
    const compiler = getCompiler({
      scale: 0.5,
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });
});
