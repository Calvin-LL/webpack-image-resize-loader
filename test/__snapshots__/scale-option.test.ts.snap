// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"scale" option should work with 0.5: result 1`] = `"44aacd9d8d250c420d100a3fb0f1debe.jpg"`;

exports[`"scale" option should work with 1: result 1`] = `"fdaf70a107ad3cebfaf1e03ff0601c1c.jpg"`;
